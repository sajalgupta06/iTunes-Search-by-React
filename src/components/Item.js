// @flow
import React, { PropTypes } from 'react';
import { getKind, kindColorMap } from '../utils';
import type { SearchResult } from '../type';
import '../style/Item.css';

const Item = (props: SearchResult) => (
  <div className="card-wrapper">
    <div className="card">
      <div className="card-image waves-effect waves-block waves-light">
        <img
          alt="img"
          className="activator"
          src={props.artworkUrl100.replace('100x100', '1200x1200')}
        />
      </div>
      <div className="card-content">
        <span className="card-title activator grey-text text-darken-4">{props.trackName || props.collectionName}<i className="material-icons right">more_vert</i></span>
        <p>
          <a target="_blank" href={props.trackViewUrl || props.collectionViewUrl}>
            more
          </a>
          {
            getKind(props.kind).length ?
              <p className={`right kind white-text ${props.kind in kindColorMap ? `${kindColorMap[props.kind]}` : 'black'}`}>
                {getKind(props.kind)}
              </p> : null
          }
        </p>
      </div>
      <div className="card-reveal">
        <span className="card-title grey-text text-darken-4">{props.trackName || props.collectionName}<i className="material-icons right">close</i></span>
        <p>{props.longDescription || props.description || 'No description.'}</p>
      </div>
    </div>
  </div>
);

export default Item;
