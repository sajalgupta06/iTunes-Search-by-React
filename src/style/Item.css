.card-wrapper {
  display: inline-block;
	width: 300px;
}

.kind {
  padding: 2px 4px;
  font-weight: 300;
  font-size: 0.8em;
  border-radius: 2px;
}
