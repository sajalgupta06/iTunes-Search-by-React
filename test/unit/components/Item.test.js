import React from 'react';
import { shallow } from 'enzyme';
import Item from '../../../src/components/Item';

describe('<Item />', () => {
  test('Item component should renders correctly', () => {
    const props = {
      artworkUrl100: '/100x100/',
      trackName: '',
      trackViewUrl: '',
      longDescription: ''
    };
    const component = shallow(<Item {...props} />);
    expect(component).toMatchSnapshot();
  });
});
