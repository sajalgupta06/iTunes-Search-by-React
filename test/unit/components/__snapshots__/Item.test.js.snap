// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> Item component should renders correctly 1`] = `
<div
  className="card-wrapper"
>
  <div
    className="card"
  >
    <div
      className="card-image waves-effect waves-block waves-light"
    >
      <img
        alt="img"
        className="activator"
        src="/1200x1200/"
      />
    </div>
    <div
      className="card-content"
    >
      <span
        className="card-title activator grey-text text-darken-4"
      >
        <i
          className="material-icons right"
        >
          more_vert
        </i>
      </span>
      <p>
        <a
          target="_blank"
        >
          more
        </a>
      </p>
    </div>
    <div
      className="card-reveal"
    >
      <span
        className="card-title grey-text text-darken-4"
      >
        <i
          className="material-icons right"
        >
          close
        </i>
      </span>
      <p>
        No description.
      </p>
    </div>
  </div>
</div>
`;
